tests/rust_syscall_tests/src/lib.rs

```diff
@@ -0,0 +1,516 @@
+
+    if ret == 0 {
+        Ok(())
+    } else {
+        Err(-ret as i32)
+    }
+}
+pub fn readv(fd: i32, iov: *const libc::iovec, iovcnt: i32) -> Result<isize> {
+    let ret = syscall(
+        libc::SYS_readv,
+        Args {
+            arg0: fd as _,
+            arg1: iov as _,
+            arg2: iovcnt as _,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret >= 0 {
+        Ok(ret)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn read(fd: i32, buf: *mut u8, count: usize) -> Result<isize> {
+    let ret = syscall(
+        libc::SYS_read,
+        Args {
+            arg0: fd as _,
+            arg1: buf as _,
+            arg2: count,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret >= 0 {
+        Ok(ret)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn write(fd: i32, buf: *const u8, count: usize) -> Result<isize> {
+    let ret = syscall(
+        libc::SYS_write,
+        Args {
+            arg0: fd as _,
+            arg1: buf as _,
+            arg2: count,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret >= 0 {
+        Ok(ret)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn close(fd: i32) -> Result<()> {
+    let ret = syscall(
+        libc::SYS_close,
+        Args {
+            arg0: fd as _,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret == 0 {
+        Ok(())
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn is_enarx() -> bool {
+    #[allow(non_upper_case_globals)]
+    const SYS_fork: i64 = 57;
+
+    let ret = syscall(SYS_fork, Args::default()).0 as i32;
+
+    match -ret {
+        0 => exit(0),
+        libc::ENOSYS => true,
+        _ => false,
+    }
+}
+
+#[repr(u64)]
+#[non_exhaustive]
+#[derive(Debug, Copy, Clone)]
+pub enum TeeTech {
+    None = 0,
+    Sev = 1,
+    Sgx = 2,
+}
+
+pub struct TryFromIntError(pub(crate) ());
+
+impl TryFrom<u64> for TeeTech {
+    type Error = TryFromIntError;
+
+    fn try_from(value: u64) -> core::result::Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::None),
+            1 => Ok(Self::Sev),
+            2 => Ok(Self::Sgx),
+            _ => Err(TryFromIntError(())),
+        }
+    }
+}
+
+pub fn get_att(nonce: Option<&mut [u8]>, buf: Option<&mut [u8]>) -> Result<(usize, TeeTech)> {
+    const SYS_GETATT: i64 = 0xEA01;
+
+    let rax: i64;
+    let rdx: u64;
+
+    let arg1 = if let Some(ref nonce) = nonce {
+        nonce.len()
+    } else {
+        0usize
+    };
+
+    let arg0 = if let Some(nonce) = nonce {
+        nonce.as_ptr() as usize
+    } else {
+        0usize
+    };
+
+    let arg3 = if let Some(ref buf) = buf {
+        buf.len()
+    } else {
+        0usize
+    };
+
+    let arg2 = if let Some(buf) = buf {
+        buf.as_mut_ptr() as usize
+    } else {
+        0usize
+    };
+
+    unsafe {
+        asm!(
+        "syscall",
```

Comment:
This can't use the syscall function?

Reply:
It can, but copy&paste from another file :)